classes/FaviconController.php

```diff
@@ -22,26 +28,37 @@ public function index()
 
         $finalPath = $inputPath = storage_path('app/media' . $settings->favicon);
 
-        if ($settings->favicon_16) {
+        return response()->file($finalPath, [
+            'Content-Type' => 'image/x-icon',
+        ]);
+    }
+
+    public function generateManifest()
+    {
+        $settings = Settings::instance();
+
+        if ($settings->webmanifest_enabled !== "1") {
+            $controller = new Controller();
+            $controller->setStatusCode(404);
+            return $controller->run('/404');
+        }
 
-            $destinationPath = storage_path('app/initbiz/seostorm/favicon/' . dirname($settings->favicon) . '/');
-            $finalPath = $outputPath = $destinationPath . basename($settings->favicon);
+        $responseArray = [];
 
-            if (!file_exists($outputPath)) {
-                if (
-                    !File::makeDirectory($destinationPath, 0777, true, true) &&
-                    !File::isDirectory($destinationPath)
-                ) {
-                    trigger_error(error_get_last(), E_USER_WARNING);
-                }
+        if ($settings->webmanifest_enabled == true) {
```

Comment:
can you use either 'true' or '1' when checking? i line 40 you have '1', here you have 'true'. Or maybe it is for some reason?

Reply:
On line 40 when i wass pulling values from settings, what i was getting was either '1' or '0' in form of a string, i don't know the exact reason, but I had to write it this way in order for the code to work